pkg/kp/pcstore/consul_store.go

```diff
@@ -113,6 +113,50 @@ func (s *consulStore) Create(
 	return pc, nil
 }
 
+// Takes in a pod cluster with the neccessary changes to its values
+// The pod cluster that gets updated has the same id as pc.ID
+func (s *consulStore) Update(
+	pc fields.PodCluster,
+	modifyIndex uint64,
+	session Session,
+) (fields.PodCluster, error) {
+	key, err := pcPath(pc.ID)
+	if err != nil {
+		return fields.PodCluster{}, err
+	}
+
+	jsonPC, err := json.Marshal(pc)
+	if err != nil {
+		// Probably the annotations don't marshal to JSON
+		return fields.PodCluster{}, util.Errorf("Unable to marshal pod cluster as JSON: %s", err)
+	}
+
+	// the chance of the UUID already existing is vanishingly small, but
+	// technically not impossible, so we should use the CAS index to guard
+	// against duplicate UUIDs
+	success, _, err := s.kv.CAS(&api.KVPair{
+		Key:         key,
+		Value:       jsonPC,
+		ModifyIndex: modifyIndex,
+	}, nil)
+	if err != nil {
+		return fields.PodCluster{}, consulutil.NewKVError("cas", key, err)
+	}
+
+	if !success {
+		return fields.PodCluster{}, util.Errorf("Could not set pod cluster at path '%s'", key)
+	}
+
+	err = s.setLabelsForPC(pc)
+	if err != nil {
+		// TODO: what if this delete fails?
+		_ = s.Delete(pc.ID)
```

Comment:
Agreed, although this code was copy/pasted from `Create`, which should also be fixed